bin/alluxio-stop.sh

```diff
@@ -75,12 +75,21 @@ stop_proxies() {
 }
 
 stop_worker() {
-  if [[ ! -z "${cache}" ]] ; then
+  if [[ ! -z "${cache}" ]]; then
     echo "Cache directory: ${cache}"
-    mkdir -p "${cache}"
-    if [[ ${?} -ne 0 ]]; then
-      echo "Failed to create directory: ${cache}"
-      exit 2
+    if [[ -d "${cache}" ]]; then
+      echo "Directory already exists at ${cache}; overwritting its contents"
+      # recursively delete all (including hidden) files of the ramdisk
+      find "${cache}" -mindepth 1 -delete
+    elif [[ -e "${cache}" ]]; then
+      echo "Non-directory file already exists at the path ${cache}; aborting"
+      exit 1
```

Comment:
Added overwriting contents of `-c cache` directory if and only if that path is a directory

Reply:
I also don't see `-mindepth` in the [POSIX spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/find.html#tag_20_47) either but it [does exist in BSD](https://www.freebsd.org/cgi/man.cgi?find(1))

Reply:
`-delete` doesn't exist in the POSIX spec either from what I can see (but it does for BSD). Any suggestions? I wanted to remove hidden files from the directory but not delete the directory itself (can't delete the ramdisk directory).

Reply:
Is using `shopt` cross-platform compliant?
```
shopt -s    dotglob
rm -rf  "${cache}/*"
shopt -u    dotglob
```

Reply:
from my understanding, `find <dir> -mindepth 1 -delete` is just looking to delete all the files on the first level in the directory `${cache}`. You can achieve the same behavior with `rm -f ${cache}/*`

Reply:
Actually let me look into using `rsync`. Since we are already using it in the `alluxio` binary I'm going to assume it's already installed on the host.

Reply:
I think I can replicate the intended behaviour with `find <dir> -depth 1` and then `rm -rf` every directory/file listed there. The issue with `rm -rf <dir>/*` is the star-glob doesn't include hidden files.

Reply:
> Is using shopt cross-platform compliant?

It's a bash built-in. Since we assume that bash is the shell in the scripts `shopt` is safe to use across platforms. Is there a reason to use `shopt` though? I've never had to use it. I just assumed globbing always works in the shell

Reply:
nvm. I realize why we needed `dotglob`. It's to handle the hidden files